### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by edi-sipka**
For PR I implemented following: 

- Created Classroom class
- Create the has-many/belongs-to relationship between Classroom and Student.

- Created a class Book.

- Created a class Rental.

- Created the many-to-many between Person, Book and Rental.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Book
+attr_accessor :title, :author
```
Comment (zieeco) [inline book.rb]: - Hey, let's set the has-many relationship of a Book and Rental, i.e, a book has many rentals. Therefore we should add the :rentals to the attribute accessor.  `attr_accessor :title, :author, :rentals`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title,author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def rent_book(rental)
+    @rentals.push(rental)
+end
```
Comment (zieeco) [inline book.rb]: - Hey, let's improve on the `rent_book/ add_rental` method as we should be able to keep track of the date and the person who is renting the book. As this helps in establishing the relationship between the Person and a Book initiated by the Rentals, which is why the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 

```  
def add_rental(person, date)
    Rental.new(date, person, self)
  end
```

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  def add_student(student)
+    @students.push(student) unless @students.include?(student)
+    student.classroom = self
+  end
+end
+
+
+
+
+
```
Comment (zieeco) [inline classroom.rb]: - [OPTIONAL] Let's remove this added space to format or codebase well enough.

File: person.rb [hunk 4]
```diff
@@ -11,8 +11,13 @@ def initialize(age, name = 'Unknown', parent_permission = "true")
     @name = name
     @age = age 
     @parent_permission = parent_permission
+    @rentals = []
   end 
 
+  def rent_book(rental)
+    @rentals.push(rental)
+end
```
Comment (zieeco) [inline person.rb]: - Hey, let's also improve on the `rent_book/ add_rental` method as we should be able to keep track of the date and the book rented by a person. As this helps in establishing the relationship between the Person and a Book initiated by the Rentals, which is why the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
```
 def add_rental(book, date)
    Rental.new(date, self, book)
  end
```

File: student.rb [hunk 5]
```diff
@@ -8,7 +8,15 @@ def initialize(classroom, age, name = 'Unknown', parent_permission = "true")
...
+
 def play_hooky
   '¯\(ツ)/¯'
  end
-end
\ No newline at end of file
+end
+
+
+
```
Comment (zieeco) [inline student.rb]: - [OPTIONAL] Let's remove this added space to format or codebase well enough.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @edi-sipka,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice setting up the association. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
_Check the comments under the review._

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 4 above)
Comment (edi-sipka) [reply person.rb]: @zieeco Thank you so much for this tips!

**Review by ShadyShawkat — APPROVED** [review]
## Hi @edi-sipka, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**